app/src/main/java/com/github/houseorganizer/houseorganizer/house/CreateHouseholdActivity.java

```diff
@@ -36,23 +50,107 @@ protected void onCreate(Bundle savedInstanceState) {
+                    location.setLatitude(addr.getLatitude());
+                    location.setLongitude(addr.getLongitude());
+                    return location;
+                }else return null;
+
+            } catch (IOException e) {
+                Toast.makeText(this, R.string.address_network_error, Toast.LENGTH_LONG ).show();
+                return null;
+            }
+        }
+    }
+
+    private void displayMapDialog(View view, Location position, String houseName){
+        final View dialogView = LayoutInflater.from(this).inflate(R.layout.map_dialog, null);
+        AlertDialog.Builder builder = new AlertDialog.Builder(this)
+                .setTitle(R.string.house_location_title)
+                .setView(dialogView)
+                .setPositiveButton(R.string.confirm, (dialog, id) -> submitHouseholdToFirestore(view, houseName, position))
+                .setNegativeButton(R.string.no, (dialog, id) -> {
+                    Toast.makeText(this, R.string.warning_address, Toast.LENGTH_LONG).show();
+                    dialog.dismiss();
+                });
+
+        AlertDialog dialog = builder.show();
+
+        MapView mMapView;
+        MapsInitializer.initialize(this);
+
+        mMapView = dialog.findViewById(R.id.mapView);
+        assert mMapView != null;
+        mMapView.onCreate(dialog.onSaveInstanceState());
+        mMapView.onResume();// needed to get the map to display immediately
+        mMapView.getMapAsync(googleMap -> {
+            LatLng pos = new LatLng(position.getLatitude(), position.getLongitude());
+            googleMap.addMarker(new MarkerOptions().position(pos).title(houseName));
+            googleMap.getUiSettings().setZoomControlsEnabled(true);
+            googleMap.animateCamera(CameraUpdateFactory.zoomTo(15), getPositionCallback(googleMap, pos));
+        });
+    }
+
+    private GoogleMap.CancelableCallback getPositionCallback(GoogleMap map, LatLng pos){
+        return new GoogleMap.CancelableCallback() {
+            @Override
+            public void onCancel() {}
+
+            @Override
+            public void onFinish() {
+                map.animateCamera(CameraUpdateFactory.newLatLng(pos));
+            }
+        };
     }
 
     public void goToQRScan(View view){
         Intent intent = new Intent(this, QRCodeScanActivity.class);
         startActivity(intent);
     }
 
-    public void submitHouseholdToFirestore(View view){
+    public void createHouseholdButtonPressed(View view){
         TextView houseHoldNameView = findViewById(R.id.editTextHouseholdName);
-        TextView latitudeView = findViewById(R.id.editTextLatitude);
-        TextView longitudeView = findViewById(R.id.editTextLongitude);
+        TextView addressView = findViewById(R.id.editTextAddress);
+
+        if(houseHoldNameView.getText().toString().equals("") || addressView.getText().toString().equals("")){
+            Toast.makeText(this, R.string.address_fill_fields, Toast.LENGTH_LONG).show();
```

Comment:
Do you need long length for toast ?

Reply:
I am not sure what it represents it is the time or the length ? in any case it displays well for now

Reply:
Ok then go with it 👍